ui/components/component-library/icon/icon.types.ts

```diff
@@ -0,0 +1,177 @@
+import type { BoxProps } from '../../ui/box/box.d';
+import { IconColor } from '../../../helpers/constants/design-system';
+
+export enum IconSize {
+  Xs = 'xs',
+  Sm = 'sm',
+  Md = 'md',
+  Lg = 'lg',
+  Xl = 'xl',
+  Inherit = 'inherit',
+}
+
+/**
+ * The IconName enum contains all the possible icon names.
+ *
+ * Search for an icon: https://metamask.github.io/metamask-storybook/?path=/story/components-componentlibrary-icon--default-story
+ *
+ * Add an icon: https://metamask.github.io/metamask-storybook/?path=/docs/components-componentlibrary-icon--default-story#adding-a-new-icon
+ *
+ */
+
+export enum IconName {
```

Comment:
Adding this as a static enum for now until we figure out a way to automate. Previously we were storing the json in a environment variable but I don't think that will work with TypeScript. I originally tried to convert the json stored in the environment variable to an enum but because it doesn't exist before run time it wont work

Reply:
@Gudahtt am I correct in thinking that it's not possible to create an enum from the `process.env.ICON_NAMES` using something like `Object.entries` because enums need to be defined statically at compile-time where as `Object.entries` happens at run time?